File: dynamic_programming/min_cost_climbing_stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        if n <= 1:
            return 0

        prev2, prev1 = 0, 0
        for i in range(2, n + 1):
            curr = min(prev1 + cost[i - 1], prev2 + cost[i - 2])
            prev2, prev1 = prev1, curr
        return prev1

File: dynamic_programming/min_cost_climbing_stairs.rb
# @param {Integer[]} cost
# @return {Integer}
def min_cost_climbing_stairs(cost)
    n = cost.size
    return 0 if n <= 1

    prev2, prev1 = 0, 0
    (2..n).each do |i|
        curr = [prev1 + cost[i - 1], prev2 + cost[i - 2]].min
        prev2, prev1 = prev1, curr
    end
    prev1
end